### PR TITLE
Quanta refactor

### PR DESCRIPTION
**Pull request by jpbrodsky**
Two tweaks:
1. GetQuanta returns additional variables describing the variance and recombination probability calculated for the event. This is useful for comparing sigma_r between NEST and data.
2. MOLAR_MASS is now a detector property so it can be changed by detectors like nEXO rather than hardcoded into NEST.

### CODE REVIEW

**Comment by jpbrodsky** [comment]
Found an issue, will fix on Monday.

**Comment by mszydagis** [comment]
Hi Jason, there is a conflict with the master branch I'm not sure how to resolve, because usually I just approve these merge requests after reading because no conflicts. It has a problem with include/NEST/NEST.hh (Also, minor aesthetic problem: the checks fail because of redundant blank line in G4integration\NESTProc.cpp:164).

**Comment by tunnell** [comment]
Hey all,

This seems to be what's going on.

<<<<<<< QuantaRefactor
  NESTcalc(const NESTcalc&) = delete;
  NESTcalc& operator=(const NESTcalc&) = delete;
=======
  NESTcalc();
>>>>>>> master

Which of the constructors do you want?  There is a 'resolve conflicts' button that you can hit to visualize the differences.

Cheers,
Chris

**Comment by mszydagis** [comment]
Well, we'll have to wait for Jason, because I'm not sure which one he wants, and it is still morning Pacific time :) but thanks Chris.

**Comment by mszydagis** [comment]
sorry @jpbrodsky just a few more things mostly very small before I can approve and merge:
- Why 2 names Molar and MOLAR in testNEST now instead of just choosing one?
- Disappearance of e- fraction being set to 1, a line under the special NaN, etc. case in NEST.cpp
- No newline at # endif at end of file (testNEST.hh)
- Why does testNEST.hh now include TestSpectra.hh?
- Are you sure this is good change form NexONi to ExcitonRatio? Because now unclear: ratio of excitons to what? Before was obvious that Nex O(ver) Ni I thought. 
- Your call since you are on EXO but you removed the lines that set the density to 3.03 g/cc. Odd. Correct value from paper.
- Lastly, it's harmless adding the Variance output to GetQuanta, but in real life you can't compare directly because of S1 and S2 noise, so still takes work to compare to NEST (see e.g. Dahl thesis)

File: include/NEST/testNEST.hh [hunk 1]
```diff
@@ -10,5 +14,5 @@ int testNEST(VDetector* detector, unsigned long int numEvts, string type,
              double eMin, double eMax, double inField, string position, string posiMuon,
              double fPos, int seed, bool no_seed);
 
-double band[NUMBINS_MAX][6];
-double energies[3];
+
+#endif
```
Comment (jpbrodsky) [inline include/NEST/testNEST.hh]:
```suggestion
#endif

```

File: include/NEST/testNEST.hh [hunk 2]
```diff
@@ -1,3 +1,7 @@
+#ifndef __TESTNEST_H__
+#define __TESTNEST_H__ 1
+
+#include "TestSpectra.hh"
```
Comment (jpbrodsky) [inline include/NEST/testNEST.hh]:
```suggestion
```

**Pull request merged**